.github/workflows/unit-tests.yml

```diff
@@ -41,11 +42,28 @@ jobs:
           pip install ninja meson scipy
           python build_tools/check-meson-openmp-dependencies.py
 
+  retrieve-commit-message:
+    name: Retrieve the latest commit message
+    runs-on: ubuntu-latest
+    if: github.repository == 'scikit-learn/scikit-learn'
+    outputs:
+      message: ${{ steps.git-log.outputs.message }}
+    steps:
+    - uses: actions/checkout@v5
```

Comment:
I would use the same pattern we use in `wheels.yml` since we know it works:

https://github.com/scikit-learn/scikit-learn/blob/90ef87a79e959dd6969bc67c2ef1cbb65a11389d/.github/workflows/wheels.yml#L35-L39

The reason it works is likely than on a push this `${{ github.event.pull_request.head.sha }}` is empty and that if empty, the default behaviour kicks in which does the right thing and checkout the branch.